src/ray/common/ray_object.h

```diff
@@ -0,0 +1,71 @@
+#ifndef RAY_COMMON_RAY_OBJECT_H
+#define RAY_COMMON_RAY_OBJECT_H
+
+#include "ray/common/buffer.h"
+#include "ray/util/logging.h"
+
+namespace ray {
+
+/// Binary representation of a ray object. It houlds buffer pointers of data and metadata.
+/// A ray object may have both data and metadata, or only one of them.
+class RayObject {
+ public:
+  /// Create a ray object instance.
+  ///
+  /// \param[in] data Data of the ray object.
+  /// \param[in] metadata Metadata of the ray object.
+  /// \param[in] copy_data Whether this class should hold a copy of data.
+  RayObject(const std::shared_ptr<Buffer> &data, const std::shared_ptr<Buffer> &metadata,
```

Comment:
Please add a comment explaining the `has_data_copy` flag and why it's necessary.

Reply:
Done.